packages/shared/src/server/repositories/traces.ts

```diff
@@ -73,6 +72,48 @@ export const getTracesTableCount = async (props: {
   return converted.length > 0 ? converted[0].count : 0;
 };
 
+export const checkTraceExists = async (
+  projectId: string,
+  traceId: string,
+  timestamp: Date | undefined,
+  filter: FilterState,
+): Promise<boolean> => {
+  const { tracesFilter } = getProjectIdDefaultFilter(projectId, {
+    tracesPrefix: "t",
+  });
+
+  tracesFilter.push(
+    ...createFilterFromFilterState(filter, tracesTableUiColumnDefinitions),
+    new StringFilter({
+      clickhouseTable: "t",
+      field: "id",
+      operator: "=",
+      value: traceId,
+    }),
+  );
+
+  const tracesFilterRes = tracesFilter.apply();
+
+  const query = `
+    SELECT id, project_id
+    FROM traces t FINAL
```

Comment:
Using FINAL here should be acceptable, because we do not rely on any skip indexes and always provide a timestamp with the current usage pattern.